diff --git a/xla/backends/cpu/runtime/convolution_lib.h b/xla/backends/cpu/runtime/convolution_lib.h
--- a/xla/backends/cpu/runtime/convolution_lib.h
+++ b/xla/backends/cpu/runtime/convolution_lib.h
@@ -414,5 +414,5 @@
           Eigen::Index end = std::min(start + task_size, feature_group_count);
           for (Eigen::Index i = start; i < end; ++i) {
-            auto on_done = [count_down]() mutable { count_down.CountDown(); };
+            auto on_done = [count_down]() mutable { const_cast<decltype(count_down)*>(&count_down)->CountDown(); };
             auto [output, convolved] = convolve_group(i);
             output.device(*device, std::move(on_done)) = convolved;
@@ -574,5 +574,5 @@

     if (device != nullptr) {
-      auto on_done = [count_down]() mutable { count_down.CountDown(); };
+      auto on_done = [count_down]() mutable { const_cast<decltype(count_down)*>(&count_down)->CountDown(); };
       output_reshaped.device(*device, std::move(on_done)) = convolved;
     } else {
